sdk/tdf_config.go

```diff
@@ -275,6 +281,81 @@ type TDFReaderConfig struct {
 
 	schemaValidationIntensity SchemaValidationIntensity
 	kasSessionKey             ocrypto.KeyPair
+	kasAllowlist              AllowList // KAS URLs that are allowed to be used for reading TDFs
+	ignoreAllowList           bool      // If true, the kasAllowlist will be ignored, and all KAS URLs will be allowed
+}
+
+type AllowList map[string]bool
+
+func getKasAddress(kasURL string) (string, error) {
+	// default to https if no scheme is provided
+	if !strings.Contains(kasURL, schemeSeperator) {
+		kasURL = schemeHTTPS + schemeSeperator + kasURL
+	}
+	parsedURL, err := url.Parse(kasURL)
+	if err != nil {
+		return "", fmt.Errorf("cannot parse kas url(%s): %w", kasURL, err)
+	}
+	if parsedURL.Hostname() == "" {
+		return "", fmt.Errorf("no host parsed from url: %s", kasURL)
+	}
+
+	// Default to "https" if no scheme is provided
```

Comment:
Seems impossible?

Reply:
if the platform url passed in on sdk initialization doesnt have a scheme this will be triggered